app/controllers/audits_controller.rb

```diff
@@ -59,8 +65,34 @@ def content_items
     @content_items ||= search.content_items.decorate
   end
 
-  def next_item
-    @next_item ||= content_items.next_item(content_item)
+  def compute_next_item_and_page
+    # Get a list of ContentItem ids for everything on the current
```

Comment:
What are your thoughts about extracting this calculation into a class? It seems a bit complex to live in the controller.

Something along the lines of:

```ruby
iterator = @search.iterator()
content_item = iterator.next(starting: index)
```

or

```ruby
iterator = Audit::Iterator.new(@search)
content_item = iterator.next()
```

And include all the logic + tests in the `Audit::Iterator` class as a first step to isolate this behaviour?